File: algorithms/3. Longest Substring Without Repeating Characters/solution.cc
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> lastPos(256, -1);
        int result = 0, start = -1;
        for ( int i = 0, n = s.size(); i < n; ++i ) {
            start = max(start, lastPos[s[i]]);
            lastPos[s[i]] = i;
            result = max(result, i - start);
        }
        return result;
    }
};

File: algorithms/3. Longest Substring Without Repeating Characters/solution.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        result, start, lastPos = 0, -1, {}
        for i, c in enumerate(s):
            if start < lastPos.get(c, -1):
                start = lastPos.get(c, -1)
            lastPos[c] = i
            result = max(result, i - start)
        return result
